test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -106,6 +106,11 @@ class test_ucp_mmap : public ucp_test {
             modify_config("PROTO_ENABLE", "n");
         }
 
+        if (get_variant_value() == VARIANT_MAP_NONBLOCK) {
+            // ODPv1 cannot interact with DEVX objects
+            modify_config("IB_MLX5_DEVX_OBJECTS", "", SETENV_IF_NOT_EXIST);
```

Comment:
1. why needed? i would expect ODP would be reported as not supported by devx md if it's only odpv1, but still be supported by mlx5dv/verbs md
2. why SETENV_IF_NOT_EXIST and not SKIP_IF_NOT_EXIST ?

Reply:
1. `test_ucp_mmap` is UCP test which has ODP variant for all test cases, so we don't specify `DEVX_OBJECTS` for all test cases it would use DEVX MD by default but without ODP support (if we are talking about ODPv1 which is the only option now). So in that general case without setting this variable ODP flag would be just ignored which is probably not what we want to test right here.
2. AFAIR `SETENV_IF_NOT_EXIST` wouldn't work since at that point it has access only to UCP config so we need to set ENV variable so it would be taken into account during UCT initialization. I will double check it.

Reply:
2. there is code in UCP to forward config to UCT

Reply:
2. Are you sure that we really have the code which propagates config from UCP to UCT level? I tried to find it or check by debugger which tables are affected by `ucp_config_modify_internal` but had no success. BTW why `SETENV_IF_NOT_EXIST ` looks worse than `SKIP_IF_NOT_EXIST` from your point of view?

Reply:
2. oks see the function ucp_apply_uct_config_list

Reply:
2. `ucp_test::modify_config` uses `ucp_config_modify_internal` which doesn't propagates environment to UCT layer.